internal_ws/ykcompile/src/lib.rs

```diff
@@ -1358,20 +1366,26 @@ impl TraceCompiler {
         // Reserved stack space for spilled locals during execution.
         let soff = self.stack_builder.size();
         // Reserved memory on the stack to spill live locals to during a guard failure.
-        let live_off = REG_POOL.len() * 8;
+        let live_off = REG_POOL.len() * usize::try_from(QWORD_REG_SIZE).unwrap();
 
         // As we'll be making calls to other functions using the Sys-V 64 ABI, we need to make sure
-        // the stack is aligned to 16 bytes before the call instruction. Note that the call
-        // instruction itself will push the return address onto the stack, so at the beginning of
-        // the trace the stack will have an extra offset of 8 bytes.
+        // the stack is correctly aligned. We must ensure that the stack pointer is 16-byte aligned
+        // immediately before the call instruction is executed.
+        //
+        // FIXME The following is not going to work in the long run, as we will have to push to the
+        // stack in order to set up arguments for calls to functions with >6 args. We will have to
+        // align at each call site.
+        //
         // Instead of adjusting the alignment before each call instruction, we do this once at the
         // beginning of the trace, since we won't be misaligning the stack during the trace
         // (push/pop is allowed as long as there's no call inbetween them). We also need to
```

Comment:
Maybe add after `there's no call inbetween them).`: `Note however that if we need to compile a call that requires stack arguments the stack needs to be re-aligned`. Alternatively, we can leave a comment in `c_call` like "Note, that even though the trace is already 16-byte aligned, that calls requiring stack arguments need to re-align the stack."

Reply:
I'm going to propose something a little more brutal. Watch this space.

Reply:
How's e671d1c?

Reply:
Love it!